lib/draft/feature_test_phoenix_integration.md

```diff
@@ -0,0 +1,232 @@
+
+So let's review this. There is a lot of detail in this story. Let's evaluate the different parts:
+
+```
+* Is it clear who the intended target user is for this feature? => YES
+* Is it clear what that user wants to be able to do? => YES
+* Is it clear what the main purpose is of this feature?  => PARTLY
+```
+
+It might seem like this is a very good story, and in many aspects it is. But although the last part is written in great detail, let's take a step back and think about this. What is the goal a teacher has when creating a contract. Or, in a more general sense, what is the goal of a contract? Is that goal really to have a mutual understanding? That is not really the goal, but more of an aspect of a contract. So what is the real end goal? If you really think this through, there is a simpler and more concise end goal that is actually the essence of why contracts are needed. Check the story below and you'll see what I mean:
+
+> as a teacher I would like to be able to create contracts for my students so that I don't have to worry about getting paid
+
+## So why is this distinction important?
+
+The main reason I think this distinction is important is this. When we know the true intent of what we are trying to solve, we might find other ways to solve that particular problem. To use this example, if the problem a teacher has is getting paid because they are constantly chasing their money, we could also solve that problem by creating a subscription based system. So the business owner asked for contracts because that is what teachers normally do when teaching, however that might not be the only option. For now I am going with this as I think it is the more simple approach to solve this problem. Building a subscription system is something I will probably tackle in the future, but I'd like to make sure we provide business value in small iterations and the first iteration for now will be about creating these contracts.
+
+## Let's write some Gherkin
+
+With one user story I often find myself writing a whole range of acceptance criteria using Gherkin. Let's start here with defining a few to get us going.
+
+```gherkin
+# Contract Listing
+GIVEN I have a teacher account at studentmanager
+AND I log in
+AND I have an existing student
+AND that student has a contract
+WHEN I visit the page for this student
+THEN I should see a list of contracts for this student
+AND should see the specific contract for this student
+
+# Adding a contract
+GIVEN I am looking at the details for a student
+WHEN I click on Add Contract
+THEN I should be able to enter all the details needed to create a contract 
+
+# Contract created flow
+GIVEN I Have entered all the details for a contract for a student
+WHEN I create this contract
+THEN I should be redirected back to the students overview page
+AND I should see the contract just created
```

Comment:
> AND I should see the contract just created

with

> AND I will see the contract created

I am thinking about the word "should" - maybe "will" is better - to assume that the system will work - if the system doesnt work, there is a failing spec?